### PR TITLE
[System] Replace NetworkCredential with referencesource

### PR DESCRIPTION
**Pull request by akoeplinger**
Fixes an issue where the Mono implementation wasn't converting between the SecureString and the plaintext version of the password.

Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=44109

### CODE REVIEW

File: mcs/class/System/ReferenceSources/SecureStringHelper.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	static partial class UnsafeNclNativeMethods
+	{
+        [SuppressUnmanagedCodeSecurity]
+        internal unsafe static class SecureStringHelper
+        {
+#if DEBUG
+            // this method is only called as part of an assert
+            internal static bool AreEqualValues(SecureString secureString1, SecureString secureString2)
+            {
+                IntPtr bstr1 = IntPtr.Zero;
```
Comment (marek-safar) [inline mcs/class/System/ReferenceSources/SecureStringHelper.cs]: I'd just return true

File: mcs/class/referencesource/System/net/System/Net/NetworkCredential.cs [hunk 2]
```diff
@@ -136,6 +147,7 @@ public NetworkCredential(string userName, SecureString password)
 //                GlobalLog.Print("NetworkCredential::set_Password: m_password: \"" + m_password + "\"" );
 #else //!FEATURE_PAL
                 m_password = UnsafeNclNativeMethods.SecureStringHelper.CreateSecureString(value);
+
```
Comment (marek-safar) [inline mcs/class/referencesource/System/net/System/Net/NetworkCredential.cs]: Unwanted change?

File: mcs/class/System/ReferenceSources/SecureStringHelper.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
+using System.Security;
+using System.Runtime.InteropServices;
+
+namespace System.Net
+{
+	static partial class UnsafeNclNativeMethods
+	{
+        [SuppressUnmanagedCodeSecurity]
```
Comment (marek-safar) [inline mcs/class/System/ReferenceSources/SecureStringHelper.cs]: No need to copy this attribute

File: mcs/class/System/ReferenceSources/SecureStringHelper.cs (hunk 1 above)
Comment (akoeplinger) [inline mcs/class/System/ReferenceSources/SecureStringHelper.cs]: That code isn't compiled at all, I just kept it to have a 1:1 copy of the code in referencesource.

File: mcs/class/System/ReferenceSources/SecureStringHelper.cs (hunk 3 above)
Comment (akoeplinger) [inline mcs/class/System/ReferenceSources/SecureStringHelper.cs]: same reason as mentioned in other comment.

File: mcs/class/System/ReferenceSources/SecureStringHelper.cs (hunk 1 above)
Comment (marek-safar) [inline mcs/class/System/ReferenceSources/SecureStringHelper.cs]: less copied code is better so this one could be not copied at all

**Pull request merged**